### ISSUE TITLE
jax.value_and_grad return gradient is zero when use jax.random.categotical 

### ISSUE THREAD

**Issue by WangHB0** [google/jax#17555]
### Description

I want to use 'jax.random.categorical' to sample from a probability matrix and use 'jax.value_and_grad' to calculate the gradient to refresh the probability matrix. But when use 'categorical', the gradient is zero while do not use categorical, just use probability matrix 

### Which accelerator(s) are you using?

CPU

### Additional system info

Python version: 3.9
